scripts/tests-with-thread-sanitizer.sh

```diff
@@ -1,14 +1,11 @@
 #!/bin/bash
+set -euo pipefail
 
-# When enableThreadSanitizer is enabled and ThreadSanitizer finds an issue, 
+# When enableThreadSanitizer is enabled and ThreadSanitizer finds an issue,
 # the logs only show failing tests, but don't highlight the threading issues.
 # Therefore we print a hint to find the threading issues.
-
 env NSUnbufferedIO=YES xcodebuild -workspace Sentry.xcworkspace -scheme Sentry -configuration Test -enableThreadSanitizer YES \
-    -destination "platform=iOS Simulator,OS=latest,name=iPhone 11" \
-    # The test works when having thread sanitizer not enabled, we have no clue why it fails, and threading issues will be caught
-    # even when it is disabled.
-    -skip-testing:"Sentry/SentrySessionTestsSwift/testInitWithJson_IfJsonContainsWrongValues_SessionIsNil" \
```

Comment:
l: Is this not an issue anymore?

Reply:
Not sure if it still causes an issue, the whole thread-sanitizer test fails at the moment (see PR description) and needs to be fixed before enabling. I'm suggesting (in the description above) to disable TSAN build for the time being. Therefore, we will only see if this specific test is a problem when all the other thread-sanitizer errors are fixed.

Reply:
I don't have the capacity to do that at the moment. As is on the `main` branch, the thread sanitizer doesn't work anyway so why not just merge the useful parts of this PR, i.e. fixing the scripts, and let the TSAN continue not working until someone has time to fix all the issues?

Reply:
I agree, we can merge this, created an issue to fix TSAN: https://github.com/getsentry/sentry-cocoa/issues/3200

Reply:
Ok, so please go ahead and disable the Thread Sanitizer test so we can merge this PR.

Reply:
thanks, disabled.

Reply:
The reason the thread sanitizer test was failing was that the destination was invalid. If you look at the [logs](https://github.com/getsentry/sentry-cocoa/actions/runs/5730391353/job/15551242053?pr=3197#step:7:6), xcodebuild warns that it received an invalid destination, shows a list of valid destinations (you can verify that the destination in this diff doesn't appear in the list), and says it's going with the first one. The first one happens to always be a macOS destination, so I'm guessing since the binary was compiled for iOS, it fails to launch.

In the future, please do not disable a check like this along with the rest of an unrelated PR. If there's an issue with an unrelated test, please fix or disable it in a separate PR. This is one of the most critical checks we have in CI, and it's been disabled for a while now.